### ISSUE TITLE
JsonbTypeDeserializer and JsonbTypeSerialiser JavaDoc is misleading

### ISSUE THREAD

**Issue by struberg** [eclipse-ee4j/jsonb-api#89]
JsonbTypeDeserializer and JsonbTypeSerializer JavaDoc describes that the annotations are meant to be applied to 'fields and properties' (ie setter/getter).
https://github.com/eclipse-ee4j/jsonb-api/blob/master/api/src/main/java/javax/json/bind/annotation/JsonbTypeSerializer.java#L27
https://github.com/eclipse-ee4j/jsonb-api/blob/master/api/src/main/java/javax/json/bind/annotation/JsonbTypeDeserializer.java#L28

But further down it notes that it also can be applied to types!
These are really two different models. Which one do we want to support? Or both?

a.)
```
@JsonbTypeDeserializer(StudentDeserializer.class)
public class Student {
  ...
}
```

or 
b.) 
```
public class Student {...}

public class StudyBook {
@JsonbTypeDeserializer(StudentDeserializer.class)
  private Student ownedBy;
```

In case we want to support both cases we should define that b.) wins over a.

Plus the spec is also not clear. In 4.7.2:
```
There are two ways how to register JsonbSerializer/JsonbDeserializer:
1. Using JsonbConfig::withSerializers/ JsonbConfig::withDeserializers method;
2. Annotating a type with JsonbSerializer/JsonbDeserializer annotation.
```

First, JsonbSerializer/JsonbDeserializer are _not* annotations. Did we mean @JsonbTypeDeserializer there? Btw, the same error is also being made in the JavaDoc ('The @JsonbSerializer annotation can be used with the following program elements'). This also should be JsonbTypeSerializer, etc.

**Comment by rmannibucau** [comment]
Hi Mark,

Think both is intended with the "b over a" rule.

Agre on the 4.7.2, probably a copy paste error.

**Comment by struberg** [comment]
Yes, that would probably be the best from a user pov. But where in the spec is this covered?

**Comment by rmannibucau** [comment]
Yhe "over" rule is not covered (just implicit annotation basee model rule) but you pointed out other 2 modes are in so guess we just need a sentence?

**Comment by m0mus** [comment]
I don't think it's covered in the spec in this particular case. But (if I remember well) this rue is covered in general. JavaDocs are incorrect, there is a bug for it https://github.com/eclipse-ee4j/jsonb-api/issues/70. Feel free to submit a PR.
